### PR TITLE
Qeauto 205 mt retailer start tab

### PR DESCRIPTION
**Pull request by DavidVallejosT**
Test Case:
[C1905 Submit the Sender form (required info only)](https://idt.testrail.net/index.php?/cases/view/1905)

### CODE REVIEW

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        [IDTFindBy(File = "SenderFormLocators", Locator = "TxtCity")]
+        private IWebElement TxtCity;
+
+        [IDTFindBy(File = "SenderFormLocators", Locator = "BtnSaveAndContinue")]
+        private IWebElement BtnSaveAndContinue;
+
+        [IDTFindBy(File = "SenderFormLocators", Locator = "CboState")]
+        private IWebElement CboState;
+
+        private PersonEntity PersonEntity;
```
Comment (mabalderramav) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs]: If you don't use this attribute please remove it.

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            this.Driver = WebDriverManager.Instance.GetWebDriver;
+            this.Wait = WebDriverManager.Instance.GetWebDriverWait;
+            IDTPageFactory.InitElements(this.Driver, this);
+        }
+
+        /// <summary>
+        /// Method to fill Sender form.
+        /// </summary>
+        /// <param name="senderEntity">Sender entity</param>
+        internal void FillSenderForm(SenderEntity senderEntity)
```
Comment (mabalderramav) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs]: Why this method is internal?
Also I think that could you to use Strategy Pattern here.
For Example:

> 
public void FillFrmAddNewFavorite(Dictionary<AddNewFavorite, string> dictionary)
        {
            new List<AddNewFavorite>(dictionary.Keys).ForEach(key => GetStrategyStepMap(dictionary)[key].Invoke());
        }
 private Dictionary<AddNewFavorite, Func<AddNewFavoriteComponent>> GetStrategyStepMap(Dictionary<AddNewFavorite, string> dictionary)
        {
            Dictionary<AddNewFavorite, Func<AddNewFavoriteComponent>> strategyMap = new Dictionary<AddNewFavorite, Func<AddNewFavoriteComponent>>();
            strategyMap.Add(AddNewFavorite.FavoriteDescription, () => FillTxtFavoriteDescription(dictionary[AddNewFavorite.FavoriteDescription]));
            return strategyMap;
        }

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            CommonActions.IsElementDisplayed(this.CboState);
+            CommonActions.ClickElement(this.CboState);
+            this.SelectComboBox(senderEntity.State);
+        }
+
+        /// <summary>
+        /// Method to select a value from Dropdown combo box.
+        /// </summary>
+        /// <param name="value">String value</param>
+        internal void SelectComboBox(string value)
```
Comment (mabalderramav) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs]: I think this method should be in core project.

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        {
+            By byLocator = Utils.GetByLocator("SenderFormLocators", "DdlState");
+            IList<IWebElement> list = CommonActions.GetListWebElements(byLocator);
+            IWebElement webElement = CommonActions.GetIWebElementByText(list, value);
+            CommonActions.ClickElement(webElement);
+        }
+
+        /// <summary>
+        /// Method to Click Save and Continue button.
```
Comment (mabalderramav) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs]: The description should be in third person

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        /// </summary>
+        internal void ClickSaveAndContinueButton()
```
Comment (mabalderramav) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs]: Change this method for private or public.

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        {
+            CommonActions.ClickElement(this.BtnSaveAndContinue);
+        }
+
+        /// <summary>
+        /// Method that obtain text from Sender information collapsed.
```
Comment (mabalderramav) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs]: The description should be in third person

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+        /// </summary>
+        /// <returns>String sender information collapsed</returns>
+        internal string GetCollapsedDatasFromUI()
```
Comment (mabalderramav) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs]: Change this method for private or public.

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+        {
+            By byLocator = Utils.GetByLocator("StartLocators", "LvSender");
+            IList<IWebElement> list = CommonActions.GetListWebElements(byLocator);
+            return (list[1].Text + list[2].Text).Replace(@",", string.Empty);
+        }
+
+        /// <summary>
+        /// Method that obtain datas from Sender form.
```
Comment (mabalderramav) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs]: The description should be in third person

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs (hunk 3 above)
Comment (DavidVallejosT) [reply IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs]: I will create another task for that method.

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs (hunk 2 above)
Comment (DavidVallejosT) [reply IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/StartPage.cs]: I implement Strategy Pattern, A lot of Thanks for that.

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/Start.cs [hunk 9]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
```
Comment (mabalderramav) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/Start.cs]: Improve the namespace according code conventions.

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/Start.cs [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace IDT.MoneyTransfer.Main.UI.Pages.MoneyTransfer.Start
```
Comment (mabalderramav) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/Start.cs]: add Doc

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/Start.cs [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public enum Start
```
Comment (mabalderramav) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/Start/Start.cs]: Add Doc

**Pull request merged**